spec/hydra/pcdm/models/object_spec.rb

```diff
@@ -14,6 +14,45 @@
     end
   end
 
+  context 'when aggregated by other objects' do
+
+    before(:all) do
+      @collection1 = Hydra::PCDM::Collection.create
+      @collection2 = Hydra::PCDM::Collection.create
+      @parent_object = Hydra::PCDM::Object.create
+      @object =  Hydra::PCDM::Object.create
+      @collection1.members << @object
+      @collection1.save
+      @collection1.reload
+      @collection2.members << @object
+      @collection2.save
+      @collection2.reload
+      @parent_object.members << @object
+      @parent_object.save
+      @parent_object.reload
```

Comment:
I believe that AF::AutosaveAssociation is meant to handle some of that, but the existing code in that Class is such a bad fit for this case that the whole thing would have to be rewritten.  I'm not taking that on.